### DISCUSSION TITLE
How argocd knows about the repo and cluster creds on adding labels 'argocd.argoproj.io/secret-type' on secrets

### DISCUSSION THREAD

**Discussion by kvinay2286** [argoproj/argo-cd#9662] [General]
Hi,
Sorry for this  novice query, I am still learning K8s & argocd and wanted to understand how argocd knows about the creds that we create through secrets with below labels:
argocd.argoproj.io/secret-type: repository
argocd.argoproj.io/secret-type: cluster

I checked deployments created by argocd but could not find secrets used as volume or env vars.

**Comment by rishabh625** [comment]
It uses kube api and queries api server to fetch secrets with this label

**Reply by kvinay2286** [reply]
OK thanks @[rishabh625](https://github.com/rishabh625)
